src/core/thread/network_data_service.cpp

```diff
@@ -132,38 +143,106 @@ void Manager::GetBackboneRouterPrimary(ot::BackboneRouter::BackboneRouterConfig
 
 #endif // (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
 
-Error Manager::GetNextSrpServerInfo(Iterator &aIterator, SrpServer::Info &aInfo) const
+Error Manager::GetNextDnsSrpAnycastInfo(Iterator &aIterator, DnsSrpAnycast::Info &aInfo) const
 {
-    Error error = kErrorNotFound;
+    Error             error       = kErrorNone;
+    uint8_t           serviceData = DnsSrpAnycast::kServiceNumber;
+    const ServiceTlv *tlv         = aIterator.mServiceTlv;
 
-    if (aIterator.mServiceTlv == nullptr)
+    do
     {
-        const uint8_t serviceData = SrpServer::kServiceNumber;
+        tlv = Get<Leader>().FindNextMatchingService(tlv, kThreadEnterpriseNumber, &serviceData, sizeof(serviceData));
+        VerifyOrExit(tlv != nullptr, error = kErrorNotFound);
 
-        aIterator.mServiceTlv = Get<Leader>().FindService(kThreadEnterpriseNumber, &serviceData, sizeof(serviceData));
-        VerifyOrExit(aIterator.mServiceTlv != nullptr);
-    }
-    else
+    } while (tlv->GetServiceDataLength() < sizeof(DnsSrpAnycast::ServiceData));
```

Comment:
The check here in `while()` is just to skip over any network data entry that does not follow current format (does not have the serial number in "service data"). We need such a check to make sure there are enough bytes so that we can parse the info from service data correctly. I opted to skip over such entries (rather than stop and return `ParseError`) to keep the implementation flexible for potential future changes.